server/src/main/java/org/elasticsearch/common/logging/DeprecatedMessage.java

```diff
@@ -21,24 +21,19 @@
 
 import org.elasticsearch.common.Strings;
 
-import java.util.Collections;
-import java.util.Map;
-
 /**
  * A logger message used by {@link DeprecationLogger}.
  * Carries x-opaque-id field if provided in the headers. Will populate the x-opaque-id field in JSON logs.
  */
-public class DeprecatedMessage extends ESLogMessage {
-
-    public DeprecatedMessage(String messagePattern, String xOpaqueId, Object... args) {
-        super(fieldMap(xOpaqueId), messagePattern, args);
-    }
+public class DeprecatedMessage  {
+    private static final String X_OPAQUE_ID_FIELD_NAME = "x-opaque-id";
 
-    private static Map<String, Object> fieldMap(String xOpaqueId) {
+    public static ESLogMessage of(String xOpaqueId, String messagePattern, Object... args){
```

Comment:
Shouldn't this be DeprecatedMessage ? 
If it should, then I think we are missing an important test in the logging FW to check that each type of message is routed correctly and we should create an issue for it.

Reply:
there used to be a hierarchy for these custom log messages, but not sure this is really needed. I thought of this class to just provide a convenience factory method. I might as well move it to `DeprecationLogger`. WDYT?